tools/DocsLinter/SimplifiedMarkdownDoc.cs

```diff
@@ -0,0 +1,204 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Markdig.Syntax;
+using Markdig.Syntax.Inlines;
+
+namespace DocsLinter
+{
+    /// <summary>
+    /// Simple data class that contains information related to a Markdown doc that we require for linting.
+    /// </summary>
+    public class SimplifiedMarkdownDoc
+    {
+        /// <summary>
+        /// List of links in the markdown file
+        /// </summary>
+        public List<LinkBase> Links = new List<LinkBase>();
+
+        /// <summary>
+        /// List of headings in the markdown file
+        /// </summary>
+        public List<Heading> Headings = new List<Heading>();
+
+        public SimplifiedMarkdownDoc()
+        {
+        }
+
+        /// <summary>
+        /// Constructor for automatically parsing a MarkdownDocument object from Markdig.
+        /// </summary>
+        /// <param name="markdownDoc"></param>
+        public SimplifiedMarkdownDoc(MarkdownDocument markdownDoc)
+        {
+            Links.AddRange(markdownDoc.Descendants().OfType<LinkInline>().Select(ParseLink));
+            Headings.AddRange(markdownDoc.Descendants().OfType<HeadingBlock>().Select(heading => new Heading(heading)));
+        }
+
+        /// <summary>
+        /// A helper function that parses the Markdig link object and returns the correct type of link - remote or local. 
+        /// </summary>
+        /// <param name="linkInline">The object from Markdig to parse.</param>
+        /// <returns></returns>
+        public static LinkBase ParseLink(LinkInline linkInline)
+        {
+            if (linkInline.Url.StartsWith("http") || linkInline.Url.StartsWith("www"))
```

Comment:
Probably a big fragile - what about `ftp://` or sites that don't start with `www`?

Reply:
Hmmm.. I wonder if there is a generic `IsUri` function floating around